workflow-templates/im-test-selenium-manual.yml

```diff
@@ -0,0 +1,223 @@
+#    Allows users to initiate the workflow manually.
+#
+# Frequency:
+#    - Manually triggered for on-demand testing.
+#
+# Projects to use this Template with:
+#    - Testing projects that use Selenium C#.
+#
+# TODO Prerequisites:
+#    - Add Application Insights secret (APPLICATIONINSIGHTS_CONNECTION_STRING) to every environment you are testing.
+#    - Update your Selenium solution to export test results to Application Insights using the IM.ExportToAppInsights NuGet package.  Detailed setup guidance can be found here: https://github.com/im-practices/techhub/blob/main/docs/Azure/content/Health-Checks/Selenium-UI-Health-Check.m#    
+
+
+
+name: 🧪 Run Selenium C# Tests
+run-name: 🧪 Run Selenium C# Tests on ${{ inputs.environment }} with ${{ inputs.browser }} browser
+
+concurrency:
+  group: selenium-${{ github.event_name == 'workflow_dispatch' && github.run_id || github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch-tag-sha:
+        description: Branch/Tag/SHA; The branch, tag, or SHA of the Selenium tests to run.
+        required: false
+      environment:
+        description: The environment to test against
+        required: true
+        type: choice
+        options:
+          - dev
+          - qa
+          - stage
+      browser:
+        description: Browser - What Browser to use for the UI tests.
+        type: choice
+        options:
+          - Chrome
+          - Firefox
+          - Edge
+      test-filter:
+        description: Test Filter - input the test category or name to run.
+        required: false
+        default: ''
+      enable-cache:
+        description: Enable caching of .NET packages to speed up builds.
+        required: false
+        type: boolean
+        default: true
+
+permissions:
+  id-token: write
+  contents: read
+  actions: read
+
+env:
+  DISPATCH_REF: ${{ inputs.branch-tag-sha || github.ref_name }}
+  SELENIUM_TEST_DIR: 'Selenium' # TODO: Update this to the root directory of the Selenium tests
+  SELENIUM_RESULTS_DIR: 'Selenium/TestResults' # TODO: Update this to the results output directory
+  DOTNET_VERSION: '8.x' # TODO: Specify the .NET SDK version to use
+  DOTNET_INSTALL_DIR: './.dotnet'  
+  NUGET_CACHE_PATH: '~/.nuget/packages'
+  TIMEZONE: America/Denver
+  
+jobs:
+  nuget-cache:
+    runs-on: im-linux
+    if: ${{ inputs.enable-cache }}
+    
+    outputs:
+      NUGET_CACHE_KEY: ${{ env.NUGET_CACHE_KEY }}
+
+    steps:
+      - name: Setup .NET # TODO: Delete if action runner contains the dotnet version you need.
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: ${{ env.DOTNET_VERSION }}
+
+      - name: Calculate Ref to Checkout
+        id: calculate-ref
+        shell: bash
+        run: |
+          GITHUB_REF="${{ env.DISPATCH_REF }}"
+          echo "GITHUB_REF=$GITHUB_REF" >> $GITHUB_ENV
+          echo "GITHUB_REF=$GITHUB_REF" >> $GITHUB_OUTPUT
```

Comment:
Why are you storing the value in `$GITHUB_OUTPUT`? I don't see that being used anywhere, just the environment variable you're setting on the line above.

Reply:
Thanks for catching that. I'll go ahead and remove the variable as it’s not used anywhere in the workflow.